test/common.js

```diff
@@ -55,3 +58,174 @@ tap.test('.util.validateID should validate FHIR id types', (t) => {
 
   t.end()
 })
+
+tap.test('testing include resources', (t) => {
+  const mongo = mongoFactory()
+  const common = commonFactory(mongo)
+  const testPatients = env.testPatients()
+
+  t.tearDown(() => {
+    mongo.closeDB(() => { })
+  })
+
+  t.test('should return an error when context parameter is null', (t) => {
+    t.rejects(common.includeResources(null, null), 'Invalid context parameter "null"')
+    t.end()
+  })
+
+  t.test('should return an error when context parameter is undefined', (t) => {
+    t.rejects(common.includeResources(void 0, null), 'Invalid context parameter "undefined"')
+    t.end()
+  })
+
+  t.test('should return an error when results parameter is null', (t) => {
+    t.rejects(common.includeResources({ test: true }, null), 'Invalid results parameter "null"')
+    t.end()
+  })
+
+  t.test('should return an error when results parameter is undefined', (t) => {
+    t.rejects(common.includeResources({ test: true }, void 0), 'Invalid results parameter "undefined"')
+    t.end()
+  })
+
+  t.test('should return an error when there is an invalid reference', withDB((t, db) => {
+    createTestPatient(db, (err, referencedPatient) => {
+      t.error(err)
+
+      const testContext = {
+        query: {
+          _include: [
+            `Patient:Patient/${referencedPatient.id}`, 'Patient:Patient/#2'
+          ]
+        }
+      }
+
+      t.rejects(common.includeResources(testContext, []), 'Invalid resource reference "Patient/#2"')
+      t.end()
+    })
+  }))
+
+  t.test('should include one resource in the results object', withDB((t, db) => {
+    createTestPatient(db, (err, referencedPatient) => {
+      t.error(err)
+
+      const testContext = {
+        query: {
+          _include: `Patient:Patient/${referencedPatient.id}`
```

Comment:
This is not the correct format according to the [FHIR spec](https://www.hl7.org/fhir/DSTU2/search.html#revinclude). The actual reference value will not be part of the `_include` parameter, just the name of the search parameter which references the resource. For example `MedicationOrder?_include=MedicationOrder:patient` indicates that Patient resources referenced by the patient [search parameter](https://www.hl7.org/fhir/DSTU2/medicationorder.html#search) of the MedicationOrder resource should be included.

Reply:
@bausmeier Ah ok I see, When I read it, it seemed to be a search query independent of the rest of the search request. But reading it again I see what you mean. I'll update it.